src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    //return true if is possible to run a backpage
+    private boolean askUserToDownload(JProgressBar progressBar,
+                                      Version current, Version remote, ReleaseInfo.Asset asset) throws IOException {
+        String message = current == null ? I18n.get("gui.backpage_button.download_message")
+                : I18n.get("gui.backpage_button.new_version_message");
+        message += "\n -Ver: " + (current != null ? current + " -> " : "") + remote + ", size: " + (asset.size >> 20) + "MB";
+
+        JButton download = new JButton(I18n.get("gui.backpage_button.download"));
+        Popups.Builder builder = Popups.of("Backpage browser", message)
+                .messageType(JOptionPane.QUESTION_MESSAGE)
+                .options(download, I18n.get("gui.backpage_button.cancel"));
+
+        JOptionPane optionPane = builder.build();
+        download.addActionListener(l -> optionPane.setValue(download));
+
+        if (download == builder.showSync()) {
+            if (!clearDirectory())
+                throw new DirectoryClearException(BACKPAGE_PATH);
+
+            FileUtils.createDirectories(BACKPAGE_PATH);
+            downloadBackpage(progressBar, asset);
+            writeVersionFile(remote);
+        }
+
+        return readVersionFile() != null;
+    }
+
+    // true if backpage folder is clear
+    private boolean clearDirectory() throws IOException {
+        if (Files.notExists(BACKPAGE_PATH)) return true;
+
+        try (Stream<Path> walk = Files.walk(BACKPAGE_PATH)) {
+            boolean backapgeFileAccessible = walk.sorted(Comparator.reverseOrder())
+                    .filter(path -> path.toString().contains(EXECUTABLE_NAME))
+                    .map(Path::toFile)
+                    .map(File::delete)
+                    .findFirst().orElse(true);
+
+            if (!backapgeFileAccessible) return false; // probably used by another process - cannot delete
+        }
+
+        try (Stream<Path> walk = Files.walk(BACKPAGE_PATH)) {
+            return walk.sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .allMatch(File::delete);
+        }
+    }
+
+    // true if everything downloaded successfully
+    private void downloadBackpage(JProgressBar progressBar, ReleaseInfo.Asset asset) throws IOException {
+        SwingUtilities.invokeLater(() -> {
+            progressBar.setIndeterminate(false);
+            progressBar.setMaximum(asset.size);
+        });
+
+        try (ZipInputStream zis = new ZipInputStream(new BufferedInputStream(asset.openStream()))) {
+            int downloadedBytes = 0;
+            for (ZipEntry entry = zis.getNextEntry(); entry != null; entry = zis.getNextEntry()) {
+                Path to = BACKPAGE_PATH.resolve(entry.getName());
+
+                if (entry.isDirectory())
+                    Files.createDirectories(to);
+                else {
```

Comment:
```suggestion
                if (entry.isDirectory()) {
                    Files.createDirectories(to);
                    continue;
                }
```
always add braces if it's not saving any lines at all
Also i think it'll be clearer if you un-indent the rest of this else statement